### PR TITLE
Bootfirmware imx support

### PR DESCRIPTION
**Pull request by igoropaniuk**
Please do not submit a Pull Request via github.  Our project makes use of
mailing lists for patch submission and review.  For more details please
see https://www.denx.de/wiki/U-Boot/Patches

The only exception to this is in order to trigger a CI loop on Azure prior
to posting of patches.

### CODE REVIEW

File: common/bootfirmware_info.c [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+		env_set("dt_bootfirmware_version", version);
+	} else {
+		ret = -FDT_ERR_NOTFOUND;
+	}
+
+out:
+	/* we should stop booting, if boot firmware version is not found */
+	if (CONFIG_IS_ENABLED(BOOTFIRMWARE_INFO_STRICT) && ret != 0) {
+		printf("Fail to read boot firmware info from DTB, "
+		       "abort boot as BOOTFIRMWARE_INFO_STRICT is set\n");
```
Comment (MrCry0) [inline common/bootfirmware_info.c:39]: As we don't abort booting right here, but return control to the parent, I'd display here a message like "Fail to read boot firmware info from DTB", and the message about aborting the boot where we really do this.

Comment (igoropaniuk) [reply common/bootfirmware_info.c:39]: @ricardosalveti asked to add explicit message here in original PR https://github.com/foundriesio/u-boot/pull/204#discussion_r1156218715

As this is is init function, returning non-zero automatillcay leads to boot abort.

Comment (MrCry0) [reply common/bootfirmware_info.c:39]: Ah, I missed that this is an init seq func. So the full msg is ok here :)

**Review by MrCry0 — APPROVED** [review]
LGTM

**Review by quaresmajose — APPROVED** [review]

**Pull request merged**